engine/src/main/java/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java

```diff
@@ -374,6 +374,34 @@ void passParametersToTrans( List<String> incomingFieldValues ) throws KettleExce
         .activateParams( trans, trans, this, trans.listParameters(), parameters.getVariable(), inputFieldValues, meta.getParameters().isInheritingAllVariables() );
   }
 
+  /**
+   * Search through the provided list for a specified value while ignoring case
+   * @param list list of Strings to be searched
+   * @param value String to search for
+   * @return true if the value is found, false otherwise
+   */
+  private boolean listContainsIgnoreCase( final List<String> list, final String value ) {
+    return list.stream().anyMatch( s -> s.equalsIgnoreCase( value ) );
+  }
+
+  /**
+   * Get the index of the specified value in the list, ignoring case.  Should only be called
+   * after verifying that the value exists first using listContainsIgnoreCase!
```

Comment:
Don't think this comment is necessary, ArrayList.indexOf() also returns -1 if not in the list

Reply:
See below.